multithreaded/Dfit_2D_main.cpp

```diff
@@ -289,73 +378,117 @@ int main(int argc, char *argv[])
-}
-std::function<double(const double*)> wrap_chi2(const std::vector<PDFInterface*>& D_PDFs_get, const std::vector<PDFInterface*>& B_PDFs_get, std::vector<std::pair<double, double>> vect_2D, const std::vector<std::vector<double>>& MC_MD, const std::vector<std::vector<double>>& dMC_MD, const std::vector<std::vector<double>>& MC_MB, const std::vector<std::vector<double>>& dMC_MB, const std::vector<std::pair<int, int>>& replaceIndexVect){
+		// Print the fit results
+		TString path, path1;
+	        if (Config::binned){
+			path1 = "results_binned100kMU_2";
+			path = TString::Format("results_binned100kMU_2/fit2D_%d", itry);
+		}
+		else{ 
+			path1 = "results_unbinned100kMU_2";
+			path = TString::Format("results_unbinned100kMU_2/fit2D_%d", itry);
+		}			
+		gSystem->Exec(TString::Format("mkdir %s", path1.Data()).Data());
+		gSystem->Exec(TString::Format("mkdir %s", path.Data()).Data());
+		std::ofstream results(TString::Format("%s/results_%d_%d.txt", path.Data(), Config::sign, int_choose_fit));
+		std::cout<<std::setprecision(25);
+		results << min->Status() << "  " << min->MinValue() << std::endl;
+		std::cout<<std::setprecision(10);
+		if (min->Status()!=0){// || min->MinValue()>24.8e6){
+			std::cout << "Bad status of fit "<< min->Status() << " chi2 is " << min->MinValue() << std::endl;
+			previous_fit = false; 
+		}else{
+			goodfit = true;
+			previous_fit = true;
+			start_scratch = false;
+		}
+		for (int i = 0; i < n_all; i++)
+		//for (int i = 0; i < (nvar_md + nvar_mb) * ncontr + ncontr; i++)
+		{
+			results << min->X()[i] << "  " << min->Errors()[i] << std::endl;
+			if(previous_fit) starting_point[i] = min->X()[i];
 
-        
-	
-        const int ncontr = Config::ncontr;
+		}
 
+		for (int i = 0; i < Config::ncontr; i++)
+			results << double(frac_indeces[i])/double(vect_2D.size()) <<"  " << 0.0 << std::endl;
+		
+		results.close();
+		const double *pa = &min->X()[Config::ncontr * (Config::nvar_md + Config::nvar_mb)];
+		// Calculate fractions
+		// Q: Why do we recalculate fractions? What does the minuti minimize - what is stored in `pa` ???
+		// The parameters pa[] define the fractions, we have 6 fractions but 5 independent parameters. The parametrisation is arbitrary
+ 		double frac_res[Config::ncontr];
+                frac_res[0] = abs(pa[0]);
+                frac_res[1] = abs(pa[1]);
+                frac_res[2] = abs(pa[2]);
+                frac_res[3] = abs(pa[3]);
+                frac_res[4] = abs(pa[4]);
+                frac_res[5] = abs(pa[5]);
 	
-	
-
-
-	double minDM = Config::minDM;
-        double maxDM = Config::maxDM;
-        double minBMcorr = Config::minBMcorr;
-        double maxBMcorr = Config::maxBMcorr;
-        const int nvar_md = Config::nvar_md;
-        const int nvar_mb = Config::nvar_mb;
-
+		if (min){
+			delete min;
+		}
+		for (int i=0; i<Config::ncontr; i++)
+			std::cout << frac_res[i] << "  frac" << i << "  " << double(frac_indeces[i])/double(vect_2D.size()) << std::endl;
+	}
+	itry++;
```

Comment:
To zakłada chyba że chcesz robić powtórkę jeśli którykolwiek z fitów w "łańcuszku" nie jest good, tak?